gensim/models/keyedvectors.py

```diff
@@ -544,7 +545,12 @@ def most_similar(self, positive=None, negative=None, topn=10, restrict_vocab=Non
                     all_words.add(self.vocab[word].index)
         if not mean:
             raise ValueError("cannot compute similarity with no input")
-        mean = matutils.unitvec(array(mean).mean(axis=0)).astype(REAL)
+        if last is None:
+            mean = matutils.unitvec(array(mean).mean(axis=0)).astype(REAL)
+        else:
+            if (len(positive)+len(negative))%2 != 0:
+                raise ValueError("wrong input word analogies.. one or more words are missing..")            
+            mean = matutils.unitvec(array(mean).mean(axis=0)/2).astype(REAL)+self.word_vec(last, use_norm=True)
```

Comment:
`mean` not unit length, looks like a bug

Reply:
btw, if `last` is missing in vocab then what happens?

Reply:
Unit length used. Fixed. Thanks!

last is always considered to be for the set based case, the first word of the last pair.